app/auth/views.py

```diff
@@ -90,30 +106,95 @@ def send_signup_email():
                 user_type=user_type
             )
-            return Response(
-                status=200,
-                headers=None,
-                response={"Email invite sent successfully"}
-            )
+            return jsonify(
+                email_address=email_address,
+                message="Email invite sent successfully"
+            ), 200
 
         except EmailError:
-            return Response(
-                status=400,
-                headers=None
-            )
+            return jsonify(
+                    email_address=email_address,
+                    message='An error occured when trying to send an email'
+                ), 500
 
     else:
-        return Response(
-                status=400,
-                headers=None
-            )
+        return jsonify(
+                email_address=email_address,
+                message='An error occured when trying to send an email'
+            ), 400
 
 
 @auth.route('/logout', methods=['GET'])
 @login_required
 def logout():
     logout_user()
-    return Response(
-        status=200,
-        headers=None
-    )
+    return jsonify(message='The user was logged out successfully'), 200
+
+
+@auth.route('/signup/validate-invite/<string:token>', methods=['GET'])
+def validate_invite_token(token):
+    try:
+        data = decode_creation_token(token.encode())
+        return jsonify(data)
+
+    except InvalidToken:
+        return jsonify(message='The token provided is invalid. It may have expired'), 400
+
+
+@auth.route('/signup/createuser', methods=['POST'])
+def submit_create_account():
+    try:
+        json_payload = get_json_from_request()
+        json_has_required_keys(json_payload, ['name', 'email_address', 'password', 'user_type'])
+        user_type = json_payload.get('user_type')
+        name = json_payload.get('name')
+        email_address = json_payload.get('email_address')
+        shared_application_id = json_payload.get('application_id', None)
+
+    except ValueError:
+        return jsonify(message='One or more required args were missing from the request'), 400
+
+    user = User.query.filter(
+        User.email_address == email_address.lower()).first()
+
+    if user is not None:
+        return jsonify(
+            application_id=user.application_id,
+            message="A user with the email address '{}' already exists".format(email_address)
+        ), 409
+
+    if user_type == "seller":
+        if shared_application_id is None:
+            application = create_application(email_address=email_address, name=name)
+            json_payload['application_id'] = application.id
+
+            if not application:
+                return jsonify(message='Application creation failed for this seller')
+
+        else:
+            json_payload['application_id'] = shared_application_id
+
+    try:
+        user = create_user(data=json_payload)
+
+        # create_user returns either error_message:str or user:Userinstance
```

Comment:
did you consider raising a custom exception rather than returning a str? Then you could just put this is in a try/except